src/com/sittinglittleduck/DirBuster/Worker.java

```diff
@@ -573,20 +321,176 @@ else if(work.getBaseCaseObj().isUseRegexInstead())
             }
             finally
             {
-                if(httpget != null)
-                {
-                    httpget.releaseConnection();
-                }
+            	if (httpMethod != null){
+            		httpMethod.releaseConnection();
+            	}
+            }
+        }
 
-                if(httphead != null)
-                {
-                    httphead.releaseConnection();
-                }
+    }
+    
+    private HttpMethodBase getHttpMethod (String method, String url) {
+    	return method.equalsIgnoreCase("HEAD") ? new HeadMethod(url) : new GetMethod(url);
```

Comment:
It only works for HEAD and GET because that's what this class uses. It could be easily expanded by using other methods if needed